From b9032a4ba3ff5622d3d38a560ca0a670c2c6ff3c Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Tue, 23 Jan 2024 07:59:04 -0800
Subject: [PATCH 1/3] ubuntu fix to disable backteace build

in ubuntu 39, the backtrace.h is found under gcc
and this causes later error that backtrace.h not found
when trying to build the backtrace support.w

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 src/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0c15c4d..ee72c83 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -130,13 +130,15 @@ set_target_properties(util PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_include_directories(util
   PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/util)
 
-check_include_file(backtrace.h BACKTRACE_H)
+#check_include_file(backtrace.h BACKTRACE_H)
 if(BACKTRACE_H)
+  message(STATUS "backtrace_h found: ${BACKTRACE_H}")
   target_compile_definitions(util PRIVATE HAVE_BACKTRACE_H)
   find_library(BACKTRACE_LIB "backtrace" ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES})
 endif()
 
 if(BACKTRACE_LIB)
+  #message(FATAL_ERROR "backtrace lib found: ${BACKTRACE_LIB}")
   target_compile_definitions(util PRIVATE ENABLE_BACKTRACE)
   target_link_libraries(util PRIVATE ${BACKTRACE_LIB})
 endif()
-- 
2.41.3

